addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2358,6 +2358,25 @@ export class OdooEditor extends EventTarget {
         const { anchorNodeOid, anchorOffset, focusNodeOid, focusOffset } = selection || {};
         const wasCollapsed =
             !selection || (focusNodeOid === anchorNodeOid && focusOffset === anchorOffset);
+        // preserve the isolation when deleting inside of an isolated link
+        const anchorNode = this.idFind(anchorNodeOid);
+        const focusNode = this.idFind(focusNodeOid) || anchorNode;
+        let isolatedLinkAnchor;
+        if (anchorNode === focusNode) {
+            isolatedLinkAnchor = closestElement(anchorNode, 'a[contentEditable="true"]');
+        }
+        const restoreIsolatedLink = () => {
+            if (
+                isolatedLinkAnchor &&
+                !isolatedLinkAnchor.querySelector('div') &&
+                !closestElement(anchorNode, '.o_not_editable')
```

Comment:
Why it does not apply if it is within a .o_not_editable as the link is itself editable?

Reply:
Not relevant anymore, but just FYI : 
this code was copied from odooEdior.js around line:2980.